### ISSUE TITLE
Autosuggestion support in Windows

### ISSUE THREAD

**Issue by MayuraVerma** [keymanapp/keyman#344]
Does keyman 10 support autosuggestion for Windows

**Comment by mcdurdin** [comment]
No, not at this time. This is slated for a future release across all platforms.

**Comment by MayuraVerma** [comment]
@mcdurdin Thanks for the response

Is it planned for 10 or further than that.

Because 10 itself is still in alpha state.

I see some dictionary files in this build. I am curious. If we have some code to test or try, I can offer my assistance.

**Comment by mcdurdin** [comment]
@MayuraVerma autosuggestion is not planned for version 10. The version 10 feature set is relatively fixed now (I hope we can provide an update on that soon) so it will be in a future version. We do welcome suggestions especially around design and will be looking for dictionaries and more in the future.

**Comment by MayuraVerma** [comment]
@mcdurdin I know dictionary for Kannada language with open license. I can upload. Please provide me instructions.

Regarding the design, please point me to the right code. I can start working on it.

**Comment by mcdurdin** [comment]
At this point we don't have any code to start working on...  we won't be ready to start thinking about it until after the 10.0 round. If you wanted to put some resources into a public github repo of your own, with license details, and post a link to the repo in this thread, then that will help greatly when we are ready to start thinking about the design.

**Comment by MayuraVerma** [comment]
@mcdurdin I see version 10 has moved to beta version.

Also version 11 alpha is released. Please include suggestion bar feature in version 11.
Google input tools, Bhasha Indic tools (microsoft subsidiary) and Windows 10 supports many languages with suggestion bar.
https://www.google.com/inputtools/windows/
http://bhashaindia.com/Downloads/Pages/newpage.aspx

I understand Keyman will help many languages which doesn't have any keyboard so far.
But suggestion bar is a very essential  feature for any language. 

Please consider this feature.

**Comment by MayuraVerma** [comment]
Reference
https://github.com/Microsoft/Windows-universal-samples/tree/master/Samples/TextSuggestion

**Comment by mcdurdin** [comment]
We are considering predictive text services for 11.0. It's not likely we can use Microsoft's infrastructure because it is tied to Windows 10 and we need something that will apply across platform and on web. So some development required. 11.0 alpha has been created as a fork from the 10.0 code: we have not yet completed the roadmap for 11.0 -- check back in a month or so!

**Comment by MayuraVerma** [comment]
Hello Marc,

I see "windows-release-alpha-11.0.5.0" is released. Is there any news on this topic?

If we are implementing this feature, what kind of word list should we build?

do we support hunspell or plain word list with its frequency?

Thanks

**Comment by mcdurdin** [comment]
Okay, so we have now completed the roadmap for 11.0. We'll work on a blog post. There are some things in flux over the next couple of weeks that may determine whether or not predictive text lands in 11.0 or 12.0. A lot depends on available resources to do the work.

A clean frequency wordlist is a great starting point for the data that will be needed. I would suggest that plain text / csv would be good until we know more about how we will implement.

**Comment by MayuraVerma** [comment]
thanks. When is 10.0 targeted to release?
Please add predictive text in 11.0. It is the NUMBER ONE required feature. Almost all 3rd party keyboard apps have this feature in all platforms.

**Comment by mcdurdin** [comment]
As I noted above:

> A lot depends on available resources to do the work.

Without the people to do the work, we can't do it!

**Comment by MayuraVerma** [comment]
@mcdurdin can we revisit this?
Windows 10 feature update version 2004 has added text suggestion with Indic phonetic keyboards.
so, there could be api's available for developers. 
Please review.
In task bar, search bar
![image](https://user-images.githubusercontent.com/14062593/89748057-05256b80-da87-11ea-8386-3c6cd36f276c.png)
In word document, at top of the document
![image](https://user-images.githubusercontent.com/14062593/89748146-64837b80-da87-11ea-9773-3cbd030131d5.png)
In word document, at bottom of the document
![image](https://user-images.githubusercontent.com/14062593/89748175-8c72df00-da87-11ea-8271-758aac2a7b94.png)

**Comment by mcdurdin** [comment]
We still don't have the developer resources to investigate this, let alone implement, at this time, sorry.

**Comment by mcdurdin** [comment]
See also https://community.software.sil.org/t/predictive-text-in-onscreen-keyboard/4907/2

Windows: General idea would be a process running the predictions worker (running a v8 instance, probably), which then passes suggestions on to keyman.exe. But there are IPC challenges on Windows; passing any content out of Modern UI apps is pretty difficult.

macOS: For research

Linux: Also, for research.

**Comment by mcdurdin** [comment]
See also https://community.software.sil.org/t/autohotkey-ahk-does-not-recognize-keyman-bengali-inputs/5029/2
